### PR TITLE
Corrected misspellings in code comments

### PR DESCRIPTION
**Pull request by bkoelman**
Manual correction of some spelling errors, one by one.
Also changed smart single/double quotes into regular quotes (except in tests that verify for their occurrence).

### CODE REVIEW

**Comment by bkoelman** [comment]
@Pilchie Could you tag some team members for review?

**Comment by Pilchie** [comment]
@pharring @cston @VSadov

File: src/Compilers/Core/Portable/Compilation/Compilation.cs [hunk 1]
```diff
@@ -1252,7 +1252,7 @@ internal bool IsRealSigned
             {
                 // A module cannot be signed. The native compiler allowed one to create a netmodule with an AssemblyKeyFile 
                 // or Container attribute (or specify a key via the cmd line). When the module was linked into an assembly,
-                // alink would sign the assembly. So rather than give an error we just don't sign when outputting a module.
+                // a link would sign the assembly. So rather than give an error we just don't sign when outputting a module.
```
Comment (cston) [inline src/Compilers/Core/Portable/Compilation/Compilation.cs]: Should be 'alink' (the framework tool) I believe.

File: src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilerDiagnosticAnalyzer.CompilationAnalyzer.cs [hunk 2]
```diff
@@ -82,7 +82,7 @@ public CompilerDiagnostic(Diagnostic original, ImmutableDictionary<string, strin
                     _properties = properties;
                 }
 
-#pragma warning disable RS0013 // we are delegating to delegatee so it is okay here
+#pragma warning disable RS0013 // we are delegating to delegate so it is okay here
```
Comment (cston) [inline src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilerDiagnosticAnalyzer.CompilationAnalyzer.cs]: 'delegatee' was intentional. Perhaps remove 'to delegatee' entirely.

**Comment by cston** [comment]
LGTM

File: src/Compilers/Core/Portable/CodeGen/ILBuilder.cs [hunk 3]
```diff
@@ -323,7 +323,7 @@ private static void MarkReachableFromBranch(ArrayBuilder<BasicBlock> reachableBl
             {
                 // if branch is blocked by a finally, then should branch to corresponding 
                 // BlockedBranchDestination instead. Original label may not be reachable.
-                // if there are no blocking finallys, then BlockedBranchDestination returns null
+                // if there are no blocking finallies, then BlockedBranchDestination returns null
```
Comment (dpoeschl) [inline src/Compilers/Core/Portable/CodeGen/ILBuilder.cs]: Maybe "finally blocks" instead?

**Comment by VSadov** [comment]
LGTM

**Comment by bkoelman** [comment]
@VSadov I would like to address review comments, but it looks like you already merged. Can I still add commits to my branch so they get added here? If not, how to proceed?

**Comment by bkoelman** [comment]
@jaredpar I'm not sure how to proceed here. Can you help me?

**Comment by VSadov** [comment]
one way to find out :-)
I guess you can add more commits to the branch, but you may need to open another PR

**Comment by jaredpar** [comment]
@bkoelman I think the best way is to open a new PR for addressing the feedback.

**Comment by bkoelman** [comment]
Thanks for helping out guys.

I tried adding more commits to my branch. After that, Github suggested to create a new pull request, because it saw this one was already merged. Awesome!

Can someone merge #3729 for me?

**Pull request merged**